src/hotspot/share/utilities/resourceHash.hpp

```diff
@@ -274,6 +274,36 @@ class ResourceHashtableBase : public STORAGE {
     }
   }
 
+  // The argument should_remove() have the signature: bool Function(K const&, V const&).
```

Comment:
Can you make the unlink a separate change?  I think unlink with the Function argument should be an overload of unlink() and implemented like the iterate function. I don't immediately understand the additional &&s.
If it's a separate change, we can fix the other callers to use lambdas, which would be nicer.
Also, I think calling the ResourceObj destructor is another discussion - they are not typically used and I think the discussion should be somewhere else.

Reply:
I will rework it.